src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -238,28 +243,190 @@ private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, ob
             // TCP Format: 
             // tcp:<host name>\<instance name>
             // tcp:<host name>,<TCP/IP port number>
-            int portNumber = 1433;
-            string[] serverAndPortParts = fullServerName.Split(',');
 
-            if (serverAndPortParts.Length == 2)
+            const int DEFAULT_SERVER_PORT = 1433;
+
+            if (String.IsNullOrWhiteSpace(fullServerName))
             {
-                try
+                return new SNITCPHandle("localhost", DEFAULT_SERVER_PORT, timerExpire, callbackObject, parallel);
```

Comment:
Looks like Native SqlClient tries to connect to localhost if no server is specified.